src/main/kotlin/race/model/Car.kt

```diff
@@ -2,28 +2,27 @@ package race.model
 
 data class Car(
     private val _name: String,
-    private val engine: Engine = RandomEngine()
-) {
-
-    val name: String get() = _name
-
     private var _position: Int = START_POSITION
-    val position: Int get() = _position
+) {
 
-    private fun move() {
-        _position += 1
+    init {
+        require(name.length < MAXIMUM_NAME_LENGTH) {
+            "자동차 이름은 5자를 초과할 수 없다."
+        }
     }
 
-    private fun isMove() = engine.cycle() >= MOVING_NUMBER
+    val name: String get() = _name
+    val position: Int get() = _position
 
-    fun go() {
-        if (isMove()) {
-            move()
-        }
+    fun move(count: Int): Car = if (count >= MOVE_CONDITION) {
+        Car(_name = name, _position = position + 1)
```

Comment:
data class의 copy를 활용해보아요!
```suggestion
        this.copy(position = position + 1)
```

Reply:
좋은 리뷰 감사합니다.